pkgs/by-name/re/rerun/package.nix

```diff
@@ -22,28 +22,32 @@
 
 rustPlatform.buildRustPackage rec {
   pname = "rerun";
-  version = "0.13.0";
-
-  src = fetchFromGitHub {
-    owner = "rerun-io";
-    repo = "rerun";
-    rev = version;
-    hash = "sha256-HgzzuvCpzKgWC8it0PSq62hBjjqpdgYtQQ50SNbr3do=";
+  version = "0.17.0";
+  src = lib.cleanSourceWith {
+    filter = path: type: builtins.baseNameOf path != "Cargo.lock";
+    src = fetchFromGitHub {
+      owner = "rerun-io";
+      repo = "rerun";
+      rev = version;
+      sha256 = "sha256-FrUxGMBau7ak9mK4MYy1LJjmTX4prMHCYYeVXJ69mdg=";
+    };
   };
-  patches = [
-    # Disables a doctest that depends on a nightly feature
-    ./0001-re_space_view_time_series-utils-patch-out-doctests-w.patch
 
+  # patch https://github.com/NixOS/nixpkgs/issues/332957
+  postPatch = ''
+    ln -s ${./Cargo.lock} Cargo.lock
```

Comment:
Upstream provides a Cargo.lock, so we don't need to put it in Nixpkgs — we can use `cargoHash` and just put a small patch updating time in `cargoPatches`.

Reply:
Thanks, TIL `cargoPatches` exist! I didn't know about that yet. Are there any better docs on the the rust utilities other than [this](https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/rust.section.md) or grepping through nixpkgs?

Reply:
I don't think so.  The documentation you linked does mention `cargoPatches`, but not in a reference list of attributes, so I can see how you missed it.  I've found that using grep to try to find how the same problem is solved in other packages to be the best way.

Reply:
@RobWalt you beat me to it! Thanks!

I noticed the patch generated from the master branch isn't applicable to the release commit, so I wanted to look into whether it's possible to pass a `postPatch` to the FOD and use some kind of a TOML editor to reset the versions unconditionally so that the "patch" would be more generally applicable. I'm unhappy about override-ability of this kind of packages...